huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy.go

```diff
@@ -0,0 +1,458 @@
+	if err != nil {
+		return fmtp.Errorf("Unable to get the create option of the throttling policy: %s", err)
+	}
+	v, err := throttles.Create(client, instanceId, opts).Extract()
+	if err != nil {
+		return common.CheckDeleted(d, err, "Error creating HuaweiCloud throttling policy")
+	}
+	d.SetId(v.Id)
+	if policies, ok := d.GetOk("user_throttles"); ok {
+		err := addSpecThrottlingPolicies(client, policies.(*schema.Set), instanceId, d.Id(), typeUser)
+		if err != nil {
+			return fmtp.Errorf("Error creating special user throttling policy: %s", err)
+		}
+	}
+	if policies, ok := d.GetOk("app_throttles"); ok {
+		err := addSpecThrottlingPolicies(client, policies.(*schema.Set), instanceId, d.Id(), typeApplication)
+		if err != nil {
+			return fmtp.Errorf("Error creating special application throttling policy: %s", err)
+		}
+	}
+	return resourceApigThrottlingPolicyV2Read(d, meta)
+}
+
+func setApigThrottlingPolicyType(d *schema.ResourceData, pType int) error {
+	for k, v := range policyType {
+		if v == pType {
+			return d.Set("type", k)
+		}
+	}
+	return fmtp.Errorf("The member type (%d) is not supported", pType)
+}
+
+func setApigThrottlingPolicyParameters(d *schema.ResourceData, config *config.Config, resp throttles.ThrottlingPolicy) error {
+	mErr := multierror.Append(nil,
+		d.Set("region", config.GetRegion(d)),
+		d.Set("name", resp.Name),
+		d.Set("period", resp.TimeInterval),
+		d.Set("period_unit", resp.TimeUnit),
+		d.Set("max_api_requests", resp.ApiCallLimits),
+		d.Set("max_user_requests", resp.UserCallLimits),
+		d.Set("max_app_requests", resp.AppCallLimits),
+		d.Set("max_ip_requests", resp.IpCallLimits),
+		d.Set("description", resp.Description),
+		d.Set("create_time", resp.CreateTime),
+		setApigThrottlingPolicyType(d, resp.Type),
+	)
+	if mErr.ErrorOrNil() != nil {
+		return mErr
+	}
+	return nil
+}
+
+func setSpecThrottlingPolicies(d *schema.ResourceData, specThrottles []throttles.SpecThrottle) error {
+	if len(specThrottles) == 0 {
+		return nil
+	}
+	// According to the rules of append method, the maximum memory is expanded to 32,
+	// and the average waste of memory is less than the waste caused by directly setting it to 30.
+	users := make([]map[string]interface{}, 0)
+	for i, throttle := range specThrottles {
+		// The special throttling policies contain user throttles and app throttles.
+		if throttle.ObjectType == typeApplication {
+			continue
+		}
+		users = append(users, map[string]interface{}{
+			"max_api_requests":       throttle.CallLimits,
+			"throttling_object_id":   throttle.ObjectId,
+			"throttling_object_name": throttle.ObjectName,
+			"id":                     throttle.ID,
+		})
+		specThrottles = append(specThrottles[:i], specThrottles[i+1:]...)
+	}
+	if err := d.Set("user_throttles", users); err != nil {
+		return fmtp.Errorf("Error saving special user throttles to state: %s", err)
+	}
+	apps := make([]map[string]interface{}, len(specThrottles))
+	for i, throttle := range specThrottles {
```

Comment:
It is better to construct `users` and `apps` in one loop.

Reply:
thx, that's easy to understand.